src/bootstrap/builder.rs

```diff
@@ -406,47 +406,61 @@ pub enum Kind {
     Check,
     Clippy,
     Fix,
+    Format,
     Test,
     Bench,
-    Dist,
     Doc,
+    Clean,
+    Dist,
     Install,
     Run,
+    Setup,
 }
 
 impl Kind {
-    fn parse(string: &str) -> Kind {
-        match string {
-            "build" => Kind::Build,
-            "check" => Kind::Check,
-            "clippy" => Kind::Clippy,
-            "fix" => Kind::Fix,
-            "test" => Kind::Test,
-            "bench" => Kind::Bench,
-            "dist" => Kind::Dist,
-            "doc" => Kind::Doc,
-            "install" => Kind::Install,
-            "run" => Kind::Run,
-            other => panic!("unknown kind: {}", other),
-        }
-    }
-
-    fn as_str(&self) -> &'static str {
+    pub fn as_str(&self) -> &'static str {
         match self {
             Kind::Build => "build",
             Kind::Check => "check",
             Kind::Clippy => "clippy",
             Kind::Fix => "fix",
+            Kind::Format => "fmt",
             Kind::Test => "test",
             Kind::Bench => "bench",
-            Kind::Dist => "dist",
```

Comment:
nit: this reordering for `Dist` isn't necessary.

Reply:
True, it's not necessary, but I swapped Dist and Doc here (and in Kind's definition) to be consistent with the ordering used everywhere else (in flags.rs, and the main help text). The consistency seems to make sense, but I'll change it back if you prefer.

Reply:
Ah, gotcha - consistency with the help text seems good, thanks for spotting it :)